src/app/gps/qgsgpsinformationwidget.cpp

```diff
@@ -541,23 +420,90 @@ void QgsGpsInformationWidget::setConnection( QgsGpsConnection *connection )
   connected( connection );
 }
 
-void QgsGpsInformationWidget::mSpinTrackWidth_valueChanged( int value )
+void QgsGpsInformationWidget::gpsSettingsChanged()
 {
-  if ( mRubberBand )
+  updateTrackAppearance();
+  updateBearingAppearance();
+
+  QgsSettings settings;
+  int acquisitionInterval = 0;
+  if ( QgsGpsConnection::settingsGpsConnectionType.exists() )
   {
-    mRubberBand->setWidth( value );
-    mRubberBand->update();
+    acquisitionInterval = static_cast< int >( QgsGpsConnection::settingGpsAcquisitionInterval.value() );
+    mDistanceThreshold = QgsGpsConnection::settingGpsDistanceThreshold.value();
+    mBearingFromTravelDirection = QgsGpsConnection::settingGpsBearingFromTravelDirection.value();
+    mMapExtentMultiplier = static_cast< int >( QgsGpsInformationWidget::settingMapExtentRecenteringThreshold.value() );
+    mMapRotateInterval = static_cast< int >( QgsGpsInformationWidget::settingMapRotateInterval.value() );
   }
+  else
+  {
+    // legacy settings
+    acquisitionInterval = settings.value( QStringLiteral( "acquisitionInterval" ), 0, QgsSettings::Gps ).toInt();
+    mDistanceThreshold = settings.value( QStringLiteral( "distanceThreshold" ), 0, QgsSettings::Gps ).toDouble();
+    mBearingFromTravelDirection = settings.value( QStringLiteral( "calculateBearingFromTravel" ), "false", QgsSettings::Gps ).toBool();
+
+    mMapExtentMultiplier = settings.value( QStringLiteral( "mapExtentMultiplier" ), "50", QgsSettings::Gps ).toInt();
+    mMapRotateInterval = settings.value( QStringLiteral( "rotateMapInterval" ), 0, QgsSettings::Gps ).toInt();
+  }
+
+  mAcquisitionInterval = acquisitionInterval * 1000;
+  if ( mAcquisitionTimer->isActive() )
+    mAcquisitionTimer->stop();
+  mAcquisitionEnabled = true;
+
+  switchAcquisition();
 }
 
-void QgsGpsInformationWidget::trackColorChanged( const QColor &color )
+void QgsGpsInformationWidget::updateTrackAppearance()
 {
-  if ( color.isValid() )  // check that a color was picked
+  if ( !mRubberBand )
+    return;
+
+  double trackWidth = 2;
+  QColor trackColor;
+  if ( settingGpsTrackWidth.exists() )
+  {
+    trackWidth = QgsGpsInformationWidget::settingGpsTrackWidth.value();
```

Comment:
Why not allowing for a complete customization of the track line (i.e. making it a line symbol instead of only exposing width and color?)

Reply:
Done